src/Common/src/System/PasteArguments.cs

```diff
@@ -29,18 +30,35 @@ public static string Paste(IEnumerable<string> arguments, bool pasteFirstArgumen
                     //   - Parsing ends at first whitespace outside quoted region.
                     //   - No way to get a literal quote past the parser.
 
+                    bool hasQuote = false;
                     bool hasWhitespace = false;
                     foreach (char c in argument)
                     {
                         if (c == Quote)
                         {
-                            throw new ApplicationException("The argv[0] argument cannot include a double quote.");
+                            hasQuote = true;
                         }
                         if (char.IsWhiteSpace(c))
                         {
                             hasWhitespace = true;
                         }
                     }
+
+                    if (hasQuote)
+                    {
+                        // If the argv[0] contains a double-quote, we're in a pickle because there is no string that would generate such an argv[0] 
+                        // under our parsing rules.
+                        //
+                        // Right now, the only user of this method that asks for arv[0] parsing is Environment.CommandLine. Since it's passing us an arg array that came from
+                        // framework itself as a result of tokenizing a command line under these same rules, this situation should never come up. But if it ever does,
+                        // throwing an exception for this seems extreme given that it's likely that the command line is just being retrieved for logging purposes
```

Comment:
> That's a nice principle but 100% code coverage is not attainable in the real world. Some contingency paths are just not practical or cost-effective to recreate in unit tests.

This isn't about code coverage numbers. The argument being made here is that the current behavior is bad and will be bad for some important case in the future, so much so that it's worth changing now even though we've never actually seen an example of it being problematic.  To me, that's something we should have a test for before making any changes, regardless of code coverage numbers.

> Let me know what you think about that

I previously introduced the RemoteInvoke functionality as a way to consolidate the myriad of individual one-off test projects we had for functionality that needed to be invoked in a separate process.  My goal had been to just have that one helper, but if there are places such a consolidated helper doesn't work or can't be made to work, it's fine for us to introduce additional ones where needed.  We should have tests for Environment.CommandLine, so if the right way to test it is to add another helper project, we should do that.

How can we be confident in a complete rewrite of Environment.CommandLine if we have literally 0 tests invoking it (and the code coverage report at https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/Code_Coverage_Report/ would seem to confirm that's the case)?

> I don't want to invest time in that if you're just going to object to it.

I'm not trying to be a naysayer.  Anything is better than nothing, but let's do the right thing, whatever that may be.